### PR TITLE
Add MCU monitoring through ADC3 for H750 and H730 based boards

### PR DESCRIPTION
**Pull request by bugobliterator**
As promised here https://github.com/ArduPilot/ardupilot/pull/22671 , requires that PR to be merged in first.

### CODE REVIEW

**Comment by andyp1per** [comment]
@bugobliterator can  you make the merge base your other PR? Difficult to review otherwise

**Comment by bugobliterator** [comment]
@andyp1per this PR is ready for your review and testing

File: libraries/AP_HAL_ChibiOS/hwdef/scripts/STM32H750xx.py [hunk 1]
```diff
@@ -76,6 +76,10 @@
 
     'DEFINES' : {
         'HAL_HAVE_HARDWARE_DOUBLE' : '1',
+        'HAL_WITH_MCU_MONITORING' : '1',
```
Comment (andyp1per) [inline libraries/AP_HAL_ChibiOS/hwdef/scripts/STM32H750xx.py:79]: These are correct for the H730 *not* the H750 which is essentially the same as the H743

**Review by andyp1per — CHANGES_REQUESTED** [review]
Need to update the correct config file

File: libraries/AP_HAL_ChibiOS/hwdef/scripts/STM32H750xx.py (hunk 1 above)
Comment (bugobliterator) [reply libraries/AP_HAL_ChibiOS/hwdef/scripts/STM32H750xx.py:79]: In that case, should we add a separate python script for H730?

Comment (bugobliterator) [reply libraries/AP_HAL_ChibiOS/hwdef/scripts/STM32H750xx.py:79]: @andyp1per feel free to push changes to this branch. I think you have the hardware where you can test this.

Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/scripts/STM32H750xx.py:79]: 😄  @bugobliterator there is a separate python script fro H730

don't think I can push to this branch, but will try

**Comment by andyp1per** [comment]
@bugobliterator I've pushed the changes but I don't think they work because ADC3 on STM32H730 is 12bit only and it looks like the code assumes 16bit. Also ADC3 is not enabled in ChibiOS so we need to ask Giovanni about that.

**Pull request closed without merging**